New version is not compatible with noImplicitAny

To reproduce:
1. Open https://stackblitz.com/edit/angular-qcxzee
2. Download Project
3. Do `npm install`
4. Do `ng build`. No errors ✔️ 
5. Edit `tsconfig.json`: add `"noImplicitAny": true,` to `compilerOptions`
6. Do `ng build`. Build fails ❌ 

Error:
```
ERROR in src/app/modal-component.ts(39,11): error TS7022: 'modalRef' implicitly
has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.
```

7. Downgrade ng-bootstrap: `npm i @ng-bootstrap/ng-bootstrap@5.1.1`
8. Do `ng build`. No errors ✔️ 

The error is caused by this code:
```ts
const modalRef = this.modalService.open(NgbdModalContent, {
  beforeDismiss: () => modalRef.componentInstance.canDismiss(),
});
```

Which was originally inspired by https://github.com/ng-bootstrap/ng-bootstrap/issues/2904#issuecomment-452702708

--- a/src/modal/modal-ref.ts
+++ b/src/modal/modal-ref.ts
@@ -30,7 +30,7 @@ export class NgbActiveModal {
 /**
  * A reference to the newly opened modal returned by the `NgbModal.open()` method.
  */
-export class NgbModalRef<T = any> {
+export class NgbModalRef {
   private _resolve: (result?: any) => void;
   private _reject: (reason?: any) => void;
 
@@ -39,7 +39,7 @@ export class NgbModalRef<T = any> {
    *
    * When a `TemplateRef` is used as the content or when the modal is closed, will return `undefined`.
    */
-  get componentInstance(): T extends new (...args: any[]) => any? InstanceType<T>: undefined {
+  get componentInstance(): any {
     if (this._contentRef && this._contentRef.componentRef) {
       return this._contentRef.componentRef.instance;
     }

--- a/src/modal/modal.spec.ts
+++ b/src/modal/modal.spec.ts
@@ -886,7 +886,7 @@ export class WithSkipTabindexFirstFocusableModalCmpt {
 })
 class TestComponent {
   name = 'World';
-  openedModal: NgbModalRef<string>;
+  openedModal: NgbModalRef;
   show = true;
   @ViewChild('content', {static: true}) tplContent;
   @ViewChild('destroyableContent', {static: true}) tplDestroyableContent;

--- a/src/modal/modal.ts
+++ b/src/modal/modal.ts
@@ -25,7 +25,7 @@ export class NgbModal {
    *
    * Also see the [`NgbModalOptions`](#/components/modal/api#NgbModalOptions) for the list of supported options.
    */
-  open<T>(content: T, options: NgbModalOptions = {}): NgbModalRef<T> {
+  open(content: any, options: NgbModalOptions = {}): NgbModalRef {
     const combinedOptions = Object.assign({}, this._config, options);
     return this._modalStack.open(this._moduleCFR, this._injector, content, combinedOptions);
   }

